src/CMakeLists.txt

```diff
@@ -45,29 +56,32 @@ add_subdirectory(${CLM_ROOT}/src/self_tests clm_self_tests)
 add_subdirectory(unit_test_stubs)
 add_subdirectory(unit_test_shr)
 
-# Remove shr_mpi_mod from share_sources.
-# This is needed because we want to use the mock shr_mpi_mod in place of the real one
+# Remove some things from share_sources
 #
 # TODO: this should be moved into a general-purpose function in Sourcelist_utils.
-# Then this block of code could be replaced with a single call, like:
+# Then each removal could be replaced with a single call, like:
 # remove_source_file(${share_sources} "shr_mpi_mod.F90")
 foreach (sourcefile ${share_sources})
+  # Remove shr_mpi_mod from share_sources.
+  # This is needed because we want to use the mock shr_mpi_mod in place of the real one
   string(REGEX MATCH "shr_mpi_mod.F90" match_found ${sourcefile})
   if(match_found)
     list(REMOVE_ITEM share_sources ${sourcefile})
   endif()
-endforeach()
 
-# Bring in PIO, jsut because it's needed for the default ESMF library and included in other submodules like share and cmeps
```

Comment:
So does the ESMF library used already have PIO built with it -- so this wasn't needed?

Reply:
It works both ways.